Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -179,6 +181,20 @@ const std::string MatrixWorkspace::getTitle() const {
     return Workspace::getTitle();
 }
 
+/** Return a reference to the SpectrumInfo object. NOT THREAD-SAFE!
+ *
+ * Currently this method is not thread-safe, and any modifications of the
+ * instrument or instrument parameters such as mask flags will not be visible in
+ * existing SpectrumInfo references. After modification, obtain a new reference
+ * by calling this method again.
+ */
+const SpectrumInfo &MatrixWorkspace::spectrumInfo() const {
```

Comment:
I now added a safety check at this position (not pushed, since there are other local changes that would make a first review more difficult):

```cpp
  if(PARALLEL_THREAD_NUMBER != 0)
    throw std::runtime_error("MatrixWorkspace::spectrumInfo(): Thread ID is "
                             "not 0. This indicates that the method is called "
                             "in a threaded context. This is not allowed.");
```